core/admin/mailu/models.py

```diff
@@ -493,10 +507,18 @@ class Token(Base):
     ip = db.Column(db.String(255))
 
     def check_password(self, password):
-        return hash.sha256_crypt.verify(password, self.password)
+        if self.password.startswith("$5$"):
+            if hash.sha256_crypt.verify(password, self.password):
+                self.set_password(password)
+                db.session.add(self)
+                db.session.commit()
+                return True
+            return False
+        return hash.pbkdf2_sha256.verify(password, self.password)
 
     def set_password(self, password):
-        self.password = hash.sha256_crypt.using(rounds=1000).hash(password)
+        # tokens have 128bits of entropy, they are not bruteforceable
+        self.password = hash.pbkdf2_sha256.using(rounds=1).hash(password)
```

Comment:
Why should the tokens use pbkdf2_sha256 with 1 round instead of e.g. bcrypt with 1 round?

Reply:
Because it's a lot faster and there is no need for any strengthening here (128 bit tokens aren't brute-forceable regardless of the hash function)

Reply:
After benchmarking it I honestly am surprised, that it's that much faster (espeically since the minimum bcrypt rounds in passlib is 4). So nevermind then.